source/infrastructure/lib/aws-instance-scheduler-stack.ts

```diff
@@ -296,141 +296,34 @@ export class AwsInstanceSchedulerStack extends cdk.Stack {
-        type: dynamodb.AttributeType.STRING
-      },
-      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
-      removalPolicy: RemovalPolicy.DESTROY,
-      pointInTimeRecovery: true
-    })
-
-    const cfnMaintenanceWindowTable = maintenanceWindowTable.node.defaultChild as dynamodb.CfnTable
-    cfnMaintenanceWindowTable.overrideLogicalId('MaintenanceWindowTable')
-    cfnMaintenanceWindowTable.addPropertyOverride("SSESpecification", {
-      "KMSMasterKeyId": { "Ref": "InstanceSchedulerEncryptionKey" },
-      "SSEEnabled": true,
-      "SSEType": 'KMS'
-    })
-    //End instance scheduler maintenance window table dynamoDB Table reference.
-
-    //Adding all the dynamo DB references to the lambda environment variables.
-    lambdaToDynamoDb.lambdaFunction.addEnvironment('CONFIG_TABLE', cfnConfigTable.ref)
-    lambdaToDynamoDb.lambdaFunction.addEnvironment('MAINTENANCE_WINDOW_TABLE', cfnMaintenanceWindowTable.ref)
-    lambdaToDynamoDb.lambdaFunction.addEnvironment('STATE_TABLE', cfnStateTable.ref)
-
-    //Start instance scheduler database policy statement for lambda.
-
-    const dynamodbPolicy = new PolicyStatement({
-      actions: [
-        'dynamodb:DeleteItem',
-        'dynamodb:GetItem',
-        'dynamodb:PutItem',
-        'dynamodb:Query',
-        'dynamodb:Scan',
-        'dynamodb:BatchWriteItem'
-      ],
-      effect: Effect.ALLOW,
-      resources: [
-        cfnConfigTable.attrArn,
-        cfnMaintenanceWindowTable.attrArn
-      ]
+    //instance scheduler core scheduler construct reference.
+    const coreScheduler = new CoreScheduler(this, 'instance-scheduler-lambda', {
+      deploymentType: DeploymentType.SOLUTIONS_S3,
+      solutionVersion: props["solutionVersion"],
+      solutionTradeMarkName: props["solutionTradeMarkName"],
+      solutionsBucket: solutionsBucket,
+      memorySize: memorySize.valueAsNumber,
+      schedulerRole: schedulerRole,
+      environment: {
+        SCHEDULER_FREQUENCY: schedulerFrequency.valueAsString,
+        TAG_NAME: tagName.valueAsString,
+        LOG_GROUP: schedulerLogGroup.logGroupName,
+        ACCOUNT: this.account,
+        ISSUES_TOPIC_ARN: snsTopic.topicArn,
+        STACK_NAME: Aws.STACK_NAME,
+        SEND_METRICS: mappings.findInMap('TrueFalse', send.findInMap('AnonymousUsage', 'Data')),
+        SOLUTION_ID: mappings.findInMap('Settings', 'MetricsSolutionId'),
+        TRACE: mappings.findInMap('TrueFalse', trace.valueAsString),
+        ENABLE_SSM_MAINTENANCE_WINDOWS: mappings.findInMap('TrueFalse', enableSSMMaintenanceWindows.valueAsString),
+        USER_AGENT: 'InstanceScheduler-' + Aws.STACK_NAME + '-' + props.solutionVersion,
+        USER_AGENT_EXTRA: `AwsSolution/${props.solutionId}/${props.solutionVersion}`,
+        METRICS_URL: mappings.findInMap('Settings', 'MetricsUrl'),
+        STACK_ID: `${cdk.Aws.STACK_ID}`,
+        UUID_KEY: send.findInMap('ParameterKey', 'UniqueId'),
+        START_EC2_BATCH_SIZE: '5'
+      }
```

Comment:
Is there anyway we could condense these environment variables? Seems like quite a few could be benefit from defaulted values.

Reply:
maybe, but not without extensive digging and possible refactoring of the lambda itself. Personally I feel that having 1 lambda handle everything is quite bloated and leads to this sort of property overlap, but that's not something that can just be fixed in a day or two.